time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultOffsetTimeArbitrary.java

```diff
@@ -0,0 +1,205 @@
+package net.jqwik.time.internal.properties.arbitraries;
+
+import java.time.*;
+import java.time.temporal.*;
+
+import org.apiguardian.api.*;
+
+import net.jqwik.api.*;
+import net.jqwik.api.arbitraries.*;
+import net.jqwik.time.api.*;
+import net.jqwik.time.api.arbitraries.*;
+
+import static java.time.temporal.ChronoUnit.*;
+import static org.apiguardian.api.API.Status.*;
+
+@API(status = INTERNAL)
+public class DefaultOffsetTimeArbitrary extends ArbitraryDecorator<OffsetTime> implements OffsetTimeArbitrary {
+
+	private OffsetTime offsetTimeMin = OffsetTime.MIN;
+	private OffsetTime offsetTimeMax = OffsetTime.MAX;
+
+	private LocalTime timeMin = null;
+	private LocalTime timeMax = null;
+
+	private int hourMin = -1;
+	private int hourMax = -1;
+
+	private int minuteMin = -1;
+	private int minuteMax = -1;
+
+	private int secondMin = -1;
+	private int secondMax = -1;
+
+	private ZoneOffset offsetMin = null;
+	private ZoneOffset offsetMax = null;
+
+	private ChronoUnit ofPrecision = null;
+
+	@Override
+	protected Arbitrary<OffsetTime> arbitrary() {
+
+		LocalTimeArbitrary localTimes = Times.times();
+		ZoneOffsetArbitrary zoneOffsets = Times.zoneOffsets();
+
+		if (timeMin != null && timeMax != null) {
+			localTimes = localTimes.between(timeMin, timeMax);
+		}
+
+		if (hourMin != -1 && hourMax != -1) {
+			localTimes = localTimes.hourBetween(hourMin, hourMax);
+		}
+
+		if (minuteMin != -1 && minuteMax != -1) {
+			localTimes = localTimes.minuteBetween(minuteMin, minuteMax);
+		}
+
+		if (secondMin != -1 && secondMax != -1) {
+			localTimes = localTimes.secondBetween(secondMin, secondMax);
+		}
+
+		if (ofPrecision != null) {
+			localTimes = localTimes.constrainPrecision(ofPrecision);
+		}
+
+		if (offsetMin != null && offsetMax != null) {
+			zoneOffsets = zoneOffsets.between(offsetMin, offsetMax);
+		}
+
+		Arbitrary<OffsetTime> offsetTimes = Combinators.combine(localTimes, zoneOffsets).as(OffsetTime::of);
+
+		offsetTimes = offsetTimes.filter(v -> !(v.isBefore(offsetTimeMin) || v.isAfter(offsetTimeMax)));
```

Comment:
Maybe you can translate offsetTimeMin/Max into the offset for an embedded localTimeArbitrary. This should get rid of all the fitering. 

Maybe it does not really make sense at all to constrain by OffsetTime instances, since one could choose min and max to have different ZoneIds. I see no point in that. If you choose to offer min and max of LocalTime (which implicitly use the generated ZoneId) all those difficulties do not arise.

Reply:
So you mean remove offsetTimeMin and offsetTimeMax? Or what do you mean?

Reply:
Use borders of type LocalTime instead.

Reply:
This helps! Thanks for the feedback :)